src/hooks/useLoginProfile.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  CodeChallengeMethod,
+  exchangeCodeAsync,
+  makeRedirectUri,
+  refreshAsync,
+  ResponseType,
+  useAuthRequest
+} from 'expo-auth-session';
+import { useEffect, useState } from 'react';
+
+import * as appJson from '../../app.json';
+import { secrets } from '../config';
+import { addToStore, readFromStore } from '../helpers';
+
+const PROFILE_ACCESS_TOKEN = 'profileAccessToken';
+
+const namespace = appJson.expo.slug as keyof typeof secrets;
+const profile = secrets[namespace]?.profile;
+const serverUrl = profile?.serverUrl;
+
+const keycloakDiscovery = {
+  authorizationEndpoint: serverUrl + '/auth',
+  tokenEndpoint: serverUrl + '/token',
+  revocationEndpoint: serverUrl + '/revoke'
+};
+
+const CLIENT_ID = profile?.clientId;
+const CLIENT_SECRET = profile?.clientSecret;
+const SCOPES = ['openid', 'email', 'profile', 'phone', 'address'];
+
+export const useLoginProfile = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const redirectUri = makeRedirectUri({
+    scheme: appJson.expo.scheme,
+    path: 'redirect'
+  });
+
+  const [request, response, promptAsync] = useAuthRequest(
+    {
+      clientId: CLIENT_ID,
+      codeChallengeMethod: CodeChallengeMethod.S256,
+      redirectUri,
+      responseType: ResponseType.Code,
+      scopes: SCOPES,
+      usePKCE: false
+    },
+    keycloakDiscovery
+  );
+
+  useEffect(() => {
+    const handleAuthResponse = async () => {
+      if (response?.type === 'success') {
+        const { code } = response.params;
+        await exchangeToken(code);
+      }
+    };
+
+    handleAuthResponse();
+  }, [response]);
+
+  const exchangeToken = async (code: string) => {
+    try {
+      setLoading(true);
+      const tokenResult = await exchangeCodeAsync(
+        {
+          code,
+          clientId: CLIENT_ID,
+          clientSecret: CLIENT_SECRET,
+          redirectUri,
+          extraParams: {
+            grant_type: 'authorization_code'
```

Comment:
why is this hardcoded if everything else is set per variables?

Reply:
Because it is used 1 time in the code, I did not feel the need to create an extra variable.